custom_components/bhyve/binary_sensor.py

```diff
@@ -0,0 +1,80 @@
+"""Support for Orbit BHyve sensors."""
+import logging
+
+from . import BHyveDeviceEntity
+from .const import (
+    DATA_BHYVE,
+    DEVICE_FLOOD,
+    EVENT_FS_ALARM,
+    EVENT_DEVICE_IDLE,
+)
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_MOISTURE,
+    BinarySensorEntity,
+)
+from .pybhyve.errors import BHyveError
+from .util import orbit_time_to_local_time
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, _discovery_info=None):
+    """Set up BHyve sensors based on a config entry."""
+    bhyve = hass.data[DATA_BHYVE]
+
+    sensors = []
+    devices = await bhyve.devices
+    for device in devices:
+        if device.get("type") == DEVICE_FLOOD:
+            sensors.append(BHyveFloodSensor(hass, bhyve, device))
+
+    async_add_entities(sensors, True)
+
+    
+class BHyveFloodSensor(BHyveDeviceEntity):
+    """Define a BHyve sensor."""
+
+    def __init__(self, hass, bhyve, device):
+        """Initialize the sensor."""
+        name = "{0} water sensor".format(device.get("name"))
+        _LOGGER.info("Creating state sensor: %s", name)
+        super().__init__(hass, bhyve, device, name, "water", DEVICE_CLASS_MOISTURE)
+
+    def _setup(self, device):
+        self._available = device.get("is_connected", False)
+        self._state = "on" if device.get("status", {}).get("flood_alarm_status") == "alarm" else "off"
```

Comment:
This logic is duplicated here in setup as well as when parsing the websocket event data. This will increase chance of the logic getting out of sync and introduce errors in the future. I suggest creating a helper function:

```py
def _setup(self, device):
    self._available = device.get("is_connected", False)
    self._state = self._parse_status(device.get("status", {}))

def _parse_status(self, status):
    return "on" if status.get("status", {}).get("flood_alarm_status") == "alarm" else "off"
```

Reply:
Testing…

Reply:
This is working on first load, but how would I call this on a WS update?

Reply:
Working. Done